src/software/MovingAverage.h

```diff
@@ -32,6 +32,7 @@ class MovingAverage {
     void begin(uint16_t smoothFactor = 10);
     void add(T newReading);
     T get();
+    double getDouble();
```

Comment:
Why have a separate double average? You could just cast it instead.

Reply:
when "T" not a double type
https://github.com/CAP1Sup/Intellistep/blob/1ecdfbb1e95047dcbec763b583aff338406f23fa/src/software/MovingAverage.h#L102-L106
get() will return T type

Reply:
Yes, but you can just cast it as a double once it returns. That's what your function does, just that it does it right before returning instead of after

Reply:
For example: **T** is **uint16_t**
uint16_t readings[10] = { 12, 12, 9, 10, 10, 9, 11, 11, 10, 10};
runningTotal = 104.0
```
// Get the smoothed result
template <typename T>
T MovingAverage<T>::get() {
    return (runningTotal / readingsNum);
}
```
runningTotal / readingsNum == 104.0 / 10 == 10.4 (double)

get() retutns **uint16_t** == (uint16_t)10.4 == **10** <- here lossed 0.4 

get() returns 10 instead of 10.4

Reply:
![image](https://user-images.githubusercontent.com/70886343/125368545-1788e300-e383-11eb-8207-4bd8d45bd5ae.png)

Reply:
Oh, I see what you mean. I forgot that the running total is always a double.